amqp_serde/src/types.rs

```diff
@@ -670,6 +670,21 @@ mod tests {
         assert_eq!("X".to_owned(), t);
     }
 
+    #[test]
+    fn test_field_value_try_into_strings() {
+        let test_str = "test_string";
+        let test_string = String::from(test_str);
+        let test_longstr = LongStr::try_from(test_str).unwrap();
+        let exp = FieldValue::S(test_longstr.clone());
+        let exp_ref = &exp;
+        let extracted_str: &str = exp_ref.try_into().unwrap();
+        let extracted_longstr: &LongStr = exp_ref.try_into().unwrap();
+        let extracted_string: &String = exp_ref.try_into().unwrap();
+        assert_eq!(test_str, extracted_str);
+        assert_eq!(test_string, *extracted_string);
+        assert_eq!(test_longstr, *extracted_longstr);
```

Comment:
```suggestion
        assert_eq!(&test_string, extracted_string);
        assert_eq!(&test_longstr, extracted_longstr);
```

Reply:
In fact both variants do exactly the same thing. So i have no objection.

Even if we "compare references" we in fact compare the referenced values.
And the references (the raw pointers) must not be equal, while our `FieldValue` variable contains the copy of original text, not a reference to it.

Reply:
agree